src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,39 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
```

Comment:
@ArtemTeslenko 
Do you mean to create another method to implement the hide state toggle logic? Because in checklist I have this point about hide method: 3. [FUNCTION NAMING] - if the function has a name hide this function should just hide but not toggle?

Reply:
Sorry, my mistake)